### ISSUE TITLE
Decoded length is incorrect

### ISSUE THREAD

**Issue by kylebme** [awelkie/cobs.rs#15]
I'm trying to use this crate in a no_std program, which means providing a larger destination buffer than the data requires. I created some basic example code that is not giving expected results. decode_with_sentinel is returning a usize of 127, no matter what source data I provide. Here is the code I'm using:
```
    let my_string_buf = b"\x00\x11\x00\x22";
    let max_len = cobs::max_encoding_length(my_string_buf.len());
    assert!(max_len < 128);
    let mut buf = [0u8; 128];

    let len = cobs::encode_with_sentinel(
        my_string_buf,
        &mut buf,
        b'\x00');

    let cobs_buf = &buf[0..len];

    let mut decoded_dest_buf = [0u8; 128];
    let new_len = cobs::decode_with_sentinel(
        cobs_buf,
        &mut decoded_dest_buf,
        b'\x00').unwrap();
    let decoded_buf = &decoded_dest_buf[0..new_len];

    writeln!(tx, "{:?}  {:?}  {:?}", my_string_buf, cobs_buf, decoded_buf).unwrap();
```

This is the result of the writeln statement:
`[0, 17, 0, 34]  [1, 2, 17, 2, 34]  [0, 17, 0, 34, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`

I also tried this example with postcard-cobs from @jamesmunns and got the same results. I'm guessing I'm just misunderstanding how I'm supposed to use the library, but I can't see another clear way to get the decoded data length.

**Comment by jamesmunns** [comment]
Hey @kylebme, thanks for the report! I plan to update postcard + the cobs crate in the coming month or two, and I'll make sure this gets looked at in that work!

Thanks again for providing a repro case.

**Comment by jamesmunns** [comment]
Hey @kylebme, this should be fixed in `cobs` v0.2.2, which should release later tonight. I've added a test to make sure this doesn't regress:

https://github.com/jamesmunns/cobs.rs/commit/07ec54585d58b38beb9f42006967944b65496372

Thanks again for the report!
